### PR TITLE
Support unique content in shared hosted template

### PR DESCRIPTION
**Pull request by kelvin-chappell**
This enables us to use the same template for different videos, so everything for episode 1 should be there.
Hopefully I haven't broken the teaser.

/cc @Calanthe @lps88

### CODE REVIEW

**Comment by Calanthe** [comment]
Maybe you should ask a backend dev for a review as well, but it's 👍  from me.

File: commercial/app/controllers/commercial/HostedContentController.scala [hunk 1]
```diff
@@ -10,14 +10,28 @@ import views.html.hosted.{episode1Video, guardianHostedPage, teaserVideo}
 object HostedContentController extends Controller {
 
   def renderHostedPage(pageName: String) = Action { implicit request =>
-    val page = HostedPage(pageName)
+    val pageUrl = controllers.commercial.routes.HostedContentController.renderHostedPage(pageName).absoluteURL
```
Comment (regiskuckaertz) [inline commercial/app/controllers/commercial/HostedContentController.scala]: Isn't it too expensive to call this function all the time, just to get an URL?

Comment (kelvin-chappell) [inline commercial/app/controllers/commercial/HostedContentController.scala]: I don't think so.  This is only going to be used for 2 or 3 pages, before we build a capi-based solution.  It gets used a lot in views to resolve URLs.

**Comment by regiskuckaertz** [comment]
Not a backend but it all looks good 👍

File: commercial/app/controllers/commercial/HostedContentController.scala (hunk 1 above)
Comment (janua) [inline commercial/app/controllers/commercial/HostedContentController.scala]: You could make it a `lazy`!

Comment (kelvin-chappell) [inline commercial/app/controllers/commercial/HostedContentController.scala]: It's always going to be used though for every request of this page.

Comment (janua) [inline commercial/app/controllers/commercial/HostedContentController.scala]: No it's not. It's only going to be needed when the `pageName` matches the `case matches` :smile:

Comment (kelvin-chappell) [inline commercial/app/controllers/commercial/HostedContentController.scala]: Oh yes.  You're right!

Comment (kelvin-chappell) [inline commercial/app/controllers/commercial/HostedContentController.scala]: Please see #13058

**Pull request merged**